packages/next/src/server/app-render/metadata-insertion/create-server-inserted-metadata.tsx

```diff
@@ -34,7 +42,18 @@ export function createServerInsertedMetadata() {
       metadataToFlush = metadataResolver()
       const html = await renderToString({
         renderToReadableStream,
-        element: <>{metadataToFlush}</>,
+        element: (
+          <>
+            {metadataToFlush}
+            <script
+              defer
+              nonce={nonce}
+              dangerouslySetInnerHTML={{
+                __html: REINSERT_ICON_SCRIPT,
+              }}
```

Comment:
can you double check if this is actually needed? we changed React to allow a lot more stuff to be encoded properly as just string children of things like scripts and style tags. This is less safe in that we do zero escaping of the content. I know we control the script so it's only risky if we put something dangerous in the script src itself but still best maybe to avoid it if we can